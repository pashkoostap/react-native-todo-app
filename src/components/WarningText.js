import React from "react";
import { Text, StyleSheet } from "react-native";

const WarningText = ({ text }) => <Text style={styles.textStyles}>{text}</Text>;

const styles = StyleSheet.create({
  textStyles: {
    fontSize: 20,
    padding: 20,
    textAlign: "center"
  }
});

export default WarningText;
