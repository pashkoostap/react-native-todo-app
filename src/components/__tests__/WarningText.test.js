import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";

import WarningText from "../WarningText";

describe("WarningText", () => {
  it("should match the snapshot", () => {
    const text = "test";

    const rendered = renderer.create(<WarningText text={text} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it("should containt Text element with provided warning text", () => {
    const text = "test";

    const root = renderer.create(<WarningText text={text} />).root;
    const innerText = root.findByType(Text).props.children;

    expect(innerText).toBe(text);
  });
});
