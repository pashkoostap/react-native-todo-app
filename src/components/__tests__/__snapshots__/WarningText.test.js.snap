// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningText should match the snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontSize": 20,
      "padding": 20,
      "textAlign": "center",
    }
  }
>
  test
</Text>
`;
